examples/nodejs/lambda-examples/cloudwatch-metrics/infrastructure/lib/momento-metrics-stack.ts

```diff
@@ -0,0 +1,310 @@
+import * as path from 'path';
+import * as cdk from 'aws-cdk-lib';
+import {Construct} from 'constructs';
+import * as lambda from 'aws-cdk-lib/aws-lambda';
+import * as lambdaNodejs from 'aws-cdk-lib/aws-lambda-nodejs';
+import * as secrets from 'aws-cdk-lib/aws-secretsmanager';
+import {FilterPattern, LogGroup, RetentionDays} from 'aws-cdk-lib/aws-logs';
+import * as ecs from 'aws-cdk-lib/aws-ecs';
+import * as iam from 'aws-cdk-lib/aws-iam';
+import {
+  Dashboard,
+  GraphWidget,
+  MathExpression,
+  Metric,
+  Shading,
+  SingleValueWidget,
+  Stats,
+  Unit,
+} from 'aws-cdk-lib/aws-cloudwatch';
+import { DockerImageAsset } from 'aws-cdk-lib/aws-ecr-assets';
+
+enum exampleApp {
+  NodejsLambda,
+  NodejsEcs,
+  Custom
+}
+
+const stackConfig: exampleApp = exampleApp.Custom;
+const customLogGroupName = '/aws/lambda/MomentoMetricsMiddlewareCDKExample';
+
+export class MomentoMetricsStack extends cdk.Stack {
+  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
+    super(scope, id, props);
+
+    const momentoApiKeyParam = new cdk.CfnParameter(this, 'MomentoApiKey', {
+      type: 'String',
+      description: 'The Momento API key that will be used to read from the cache.',
+      noEcho: true,
+    });
+
+    const apiKeySecret = new secrets.Secret(this, 'MomentoMetricsApiKey', {
+      secretName: 'MomentoMetricsApiKey',
+      secretStringValue: new cdk.SecretValue(momentoApiKeyParam.valueAsString),
+    });
+
+    const configToLogGroupName = new Map([
+      [exampleApp.NodejsLambda, '/aws/lambda/MomentoMetricsMiddlewareCDKExample'],
+      [exampleApp.NodejsEcs, '/aws/ecs/MomentoMetricsMiddlewareCDKExample'],
+      [exampleApp.Custom, customLogGroupName]
+    ])
+    const logGroupName = configToLogGroupName.get(stackConfig);
+
+    const logGroup = new LogGroup(this, 'Logs', {
+      logGroupName: logGroupName,
+      retention: RetentionDays.ONE_DAY,
+    });
+
+    switch (stackConfig) {
```

Comment:
I would love to move all of the stuff that has to do with deploying the lambda/ecs examples to a few smaller, purpose-built functions so that they don't distract from the flow of readability for this main code.